python/pyspark/sql/connect/plan.py

```diff
@@ -712,6 +712,8 @@ def __init__(self, child: Optional["LogicalPlan"], alias: str) -> None:
 
     def plan(self, session: "RemoteSparkSession") -> proto.Relation:
         rel = proto.Relation()
+        if self._child is not None:
+            rel.subquery_alias.input.CopyFrom(self._child.plan(session))
```

Comment:
Can we have an e2e test with `DataFrame.alias` since we're here?

Reply:
Done.

BTW the e2e tests are slow. Not sure if we want to expand connect e2e test suites a lot eventually which could slow down the run job.

Reply:
The current e2e tests run about 16s on my laptop and the majority of the time is spend in starting spark. When we keep the restarts of Spark minimal, e2e tests should not be an issue when they make sense.

Reply:
That will make sense then